src/js/jsonEslint.js

```diff
@@ -30,81 +30,72 @@ fluid.defaults("fluid.lintAll.jsonEslint", {
-        eslintSequence.then(
-            function () {
-                fluid.each(eslintResults, function (singleError) {
-                    var relativePath = path.relative(that.options.rootPath, singleError.filePath);
-
-                    var fileErrorCount    = fluid.get(singleError, "errorCount");
-                    if (fileErrorCount) {
-                        that.results.invalid++;
-                        that.results.errorsByPath[relativePath] = [];
-
-                    }
-                    else {
-                        that.results.valid++;
-                    }
-
-                    fluid.each(singleError.messages, function (singleMessage) {
-                        // Our "wrapper" adds three lines of code before the first line.
-                        var adjustedLine = singleMessage.line - 3;
-                        // Our "wrapper" adds 17 characters to the first line.
-                        var adjustedColumn = adjustedLine === 1 ? singleMessage.column - 17 : singleMessage.column;
-
-                        that.results.errorsByPath[relativePath].push({
-                            line: adjustedLine,
-                            column: adjustedColumn,
-                            message: singleMessage.message
-                        });
+            });
+        } catch (err) {
+            fluid.fail(err);
+        }
+    });
+
+    var eslintSequence = fluid.promise.sequence(eslintPromises);
+    eslintSequence.then(
+        function () {
+            fluid.each(eslintResults, function (singleError) {
+                var relativePath = path.relative(that.options.rootPath, singleError.filePath);
+
+                var fileErrorCount    = fluid.get(singleError, "errorCount");
+                if (fileErrorCount) {
+                    that.results.invalid++;
+                    that.results.errorsByPath[relativePath] = [];
+
+                }
+                else {
+                    that.results.valid++;
+                }
+
+                fluid.each(singleError.messages, function (singleMessage) {
+                    // Our "wrapper" adds three lines of code before the first line.
+                    var adjustedLine = singleMessage.line - 3;
+                    // Our "wrapper" adds 17 characters to the first line.
+                    var adjustedColumn = adjustedLine === 1 ? singleMessage.column - 17 : singleMessage.column;
+
+                    that.results.errorsByPath[relativePath].push({
+                        line: adjustedLine,
+                        column: adjustedColumn,
+                        message: singleMessage.message
                     });
                 });
-                wrappedPromise.resolve(that.results);
-            },
-            wrappedPromise.reject
-        );
-    }
-    else {
-        wrappedPromise.resolve(that.results);
-    }
+            });
+            wrappedPromise.resolve(that.results);
+        },
+        wrappedPromise.reject
```

Comment:
This could probably be https://docs.fluidproject.org/infusion/development/promisesapi#fluidpromisemapsource-func ?

Reply:
Agreed.  I reviewed other uses of wrapped promises, and left them alone, as they're either wrapping non-Fluid promises or needed to catch and report low-level errors.